include/knowhere/index/index_node.h

```diff
@@ -71,6 +73,16 @@ class IndexNode : public Object {
         return Add(dataset, std::move(cfg));
     }
 
+/*
+ *@ @brief Builds the index using the provided dataset,configuration and handle.
+ */
+#ifdef KNOWHERE_WITH_CARDINAL
+    virtual Status
+    BuildAsync(const DataSetPtr dataset, std::shared_ptr<Config> cfg, const Handle* = nullptr) {
```

Comment:
I want to maintain compatibility with the existing code.

Reply:
The open-source version only supports asynchronous operations but doesn't allow the build process to be stopped proactively, whereas Cardinal supports this feature. To enable this in the open-source version, it would require a significant amount of code changes.

Reply:
> The open-source version only supports asynchronous operations but doesn't allow the build process to be stopped proactively, whereas Cardinal supports this feature. To enable this in the open-source version, it would require a significant amount of code changes.

Do you mean the change in different algos to interrupt? How about keeping the API sync and simply throw exception when the `stop` get called?